lib/src/node.dart

```diff
@@ -56,30 +68,89 @@ class StringNode extends Node {
         yamlLines.add(lines[index]);
       }
     } else {
-      var containsSingleQuote = text.contains("'");
-
-      if (context.allowUnquotedStrings &&
-          !containsSingleQuote &&
-          _isValidUnquotedString(text)) {
+      if (!context.config.forceQuotedString && isValidUnquotedString(text)) {
+        // if forceQuotedString is false and the string is valid unquoted,
+        // don't use any quote
         yamlLines.add(text);
-      } else if (!containsSingleQuote) {
-        yamlLines.add('\'$text\'');
       } else {
-        var str = text.replaceAll('\\', '\\\\').replaceAll('"', '\\"');
-        yamlLines.add('"$str"');
+        var result = text;
+        switch (context.config.quoteStyle) {
+          case QuoteStyle.singleQuote:
+            result = result.replaceAll("'", "''");
+            yamlLines.add("'$result'");
+            break;
+          case QuoteStyle.doubleQuote:
+            result = result.replaceAll('"', r'\"');
+            yamlLines.add('"$result"');
```

Comment:
With double quotes, you still need to escape any `\` character.

Reply:
Please, add a unit test for escaped characters, and if it can be read by
https://pub.dev/packages/yaml
(added as dev_dependency, only for tests).

Reply:
> With double quotes, you still need to escape any `\` character.

Now I escape the `\` char.

Reply:
> Please, add a unit test for escaped characters, and if it can be read by https://pub.dev/packages/yaml (added as dev_dependency, only for tests).

Tests are added.